server/src/main/java/org/opensearch/monitor/fs/FsInfo.java

```diff
@@ -75,16 +76,37 @@ public static class Path implements Writeable, ToXContentObject {
         long free = -1;
         long available = -1;
         long fileCacheReserved = -1;
-        long fileCacheUtilized = 0;
+        long fileCacheUtilized = -1;
 
         public Path() {}
 
+        /**
+         * Please notice that this constructor will use value of <code>0</code>
+         * for <code>fileCacheReserved</code> and <code>fileCacheUtilized</code> variables.
+         *
+         * See {@link #getFileCacheReserved()}, {@link #getFileCacheUtilized()}
+         */
         public Path(String path, @Nullable String mount, long total, long free, long available) {
+            new Path(path, mount, total, free, available, 0, 0);
+        }
+
+        @InternalApi
+        public Path(
+            String path,
+            @Nullable String mount,
+            long total,
+            long free,
+            long available,
+            long fileCacheReserved,
+            long fileCacheUtilized
+        ) {
             this.path = path;
             this.mount = mount;
             this.total = total;
             this.free = free;
             this.available = available;
+            this.fileCacheReserved = fileCacheReserved;
```

Comment:
Please add validation for `fileCacheReserved` and `fileCacheUtilized`:  `-1` && `>= 0`

Reply:
@reta You will have to give me very good reason please otherwise I am not going to do it.

I am not saying this is a bad idea (it sounds valid at first) but I would first point you to methods `FsInfo.Path#add(...)` and `FsInfo.Path#addLong(...)`. These two methods are used during the code execution and there are tests covering their application. (I did not invented the logic behind these two methods, this is something that goes a lot deeper into history of the code...)

Reply:
> @reta You will have to give me very good reason please otherwise I am not going to do it.

@lukas-vlcek I will give you 2 reasons:

1. You are accepting **new args** which where **never** set externally but adjuested (using `FsProbe.adjustForHugeFilesystems`)
2. You have a special meaning for `-1` value, anything below this value down the negative range has no meaning

Reply:
@reta Please advice how we can then implement tests in `FsProbeTests.testFsInfoOverflow()`.

You are correct that these values are not set externally, which is not true for tests.
If we accept that the disk size (like `available` variable) can overflow then IMO we need to apply the same logic to file caches as well.

Reply:
@reta Apology, the `FsInfo.Path` ctor is never used with negative arguments in tests. I can add `assert`s and it shouldn't have any negative effect on existing code in the core. Though, I would like to do it for the other arguments as well (ie. not only for `fileCache*` but also for `total`, `free` and `available`).

Reply:
> Though, I would like to do it for the other arguments as well (ie. not only for `fileCache*` but also for `total`, `free` and `available`).

@lukas-vlcek please do